spec/requests/cloud_templates_spec.rb

```diff
@@ -55,4 +55,24 @@
       expect(response).to have_http_status(:forbidden)
     end
   end
+
+  describe "DELETE /api/providers/:c_id/cloud_templates/:id" do
+    it "can delete an image" do
+      api_basic_authorize(action_identifier(:cloud_templates, :delete, :subresource_actions, :delete))
+      ems = FactoryGirl.create(:ems_cloud)
+      image = FactoryGirl.create(:template_cloud, :ext_management_system => ems)
+
+      delete(api_provider_cloud_template_url(nil, ems, image))
+      expect(response).to have_http_status(:no_content)
+    end
+
+    it "will not delete image unless authorized" do
+      api_basic_authorize
+      ems = FactoryGirl.create(:ems_cloud)
+      image = FactoryGirl.create(:template_cloud, :ext_management_system => ems)
+
+      delete(api_provider_cloud_template_url(nil, ems, image))
+      expect(response).to have_http_status(:forbidden)
+    end
+  end
```

Comment:
will need tests for the POST based deletes, single and bulk.

Reply:
Thanks for notice, Tests added